impress-org/givewp
# Feature: Add 'Additional fields' FFM recommendation to form builder

## Description

This PR inserts the Additional fields panel inside the secondary sidebar on the primary block list. The panel displays the additional fields you can use in the form builder if you install the FFM add-on. A link is also displayed  which opens a new web page with additional information.The Additional fields panel will only render when FFM is NOT active and installed. The fields are not interactive, as such they have a "locked" appearance.

To limit the Additional panel to display only on the Primary block list, a Gutenberg selector was used to check for a selectedBlock. The primary block list is only visible when a block other than ```givewp/section```  has been selected.

## Visuals

<img width="1718" alt="Screen Shot 2023-08-23 at 3 02 02 PM" src="https://github.com/impress-org/givewp/assets/75056371/9940bdd2-2ae8-48a3-b989-4656ae5df56f">

## Testing Instructions
- Add a new form.
- You will need to select a block and then open the secondary sidebar.
- Scroll to the bottom of the list to view the additional fields.
- Test the link.
- Install FormFieldManager
- View the form again and verify that it does not display.

- In the video you may notice the upgrade button text changes to a dark color after being clicked. This has since been fixed.
https://www.loom.com/share/9dc8307e041e4f1e97ddbf7d8a9a60b7?sid=60e41b30-dde1-4029-8418-3866b1a4955c

## Pre-review Checklist



-   [x] Includes update to unit tests to include new value for the view model [tests/Unit/ViewModels/FormBuilderViewModelTest.php](https://github.com/impress-org/givewp/pull/6897/files#diff-d1f7b58727a355c9bf21034aab9e1577105357b9765ac21ecbb350959c460056)
-   [x ] Reviewed by the designer (if follows a design)



---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205360468064596
